pyiceberg/partitioning.py

```diff
@@ -193,6 +197,27 @@ def partition_type(self, schema: Schema) -> StructType:
             nested_fields.append(NestedField(field.field_id, field.name, result_type, required=False))
         return StructType(*nested_fields)
 
+    def partition_to_path(self, data: Record, schema: Schema) -> str:
+        partition_type = self.partition_type(schema)
+        field_types = partition_type.fields
+
+        pos = 0
+        field_strs = []
+        value_strs = []
+        for field_name in data._position_to_field_name:
```

Comment:
Maybe use `enumerate` here to avoid having to increment the `pos`?

Reply:
Yes that is much better, thank you!